lib/Refactoring/Refactoring.cpp

```diff
@@ -8688,7 +8686,19 @@ bool RefactoringActionExpandMacro::performChange() {
     rewrittenBuffer = adjustMacroExpansionWhitespace(
         generatedInfo->kind, rewrittenBuffer, scratchBuffer);
 
-    EditConsumer.accept(SM, originalSourceRange, rewrittenBuffer);
+    SourceFile *originalFile =
+        MD->getSourceFileContainingLocation(originalSourceRange.getStart());
+    StringRef originalPath;
+    if (originalFile->getBufferID().hasValue() &&
+        TheFile->getBufferID() != originalFile->getBufferID()) {
```

Comment:
It's a little confusing what is `originalFile` and what is `TheFile` here. Maybe some comment with a example helps?

Reply:
Yeah that's fair, will do.